[BUG] Release build with mvn option `-P source-javadoc` FAILED

**Describe the bug**
When running release build, we need to build `javadoc.jar` and `source.jar` with mvn option  `-P source-javadoc` (the only diff from the nightly build) , but build FAILED in this line https://github.com/NVIDIA/spark-rapids/blob/branch-22.02/jenkins/spark-nightly-build.sh#L30,  error output as below:

```
[ERROR] [ERROR] Could not find the selected project in the reactor: dist @
[ERROR] Could not find the selected project in the reactor: dist -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException

```

Quick repro
```
cd spark-rapids
mvn help:evaluate -q -pl dist  -P mirror-apache-to-urm,source-javadoc  -Dcuda.version=cuda11 -DforceStdout -Dexpression=project.artifactId
```

```
 mvn clean package -pl dist  -P mirror-apache-to-urm,source-javadoc  -Dcuda.version=cuda11 -DforceStdout -Dexpression=project.artifactId
```

Note:  

Build **PASS** when we remove `-P source-javadoc`

Build **PASS** if we revert root pom.xml changed in the PR : https://github.com/NVIDIA/spark-rapids/pull/4529/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8

--- a/jenkins/spark-nightly-build.sh
+++ b/jenkins/spark-nightly-build.sh
@@ -24,7 +24,7 @@ export M2DIR="$WORKSPACE/.m2"
 
 DIST_PL="dist"
 function mvnEval {
-    mvn help:evaluate -q -pl $DIST_PL $MVN_URM_MIRROR -Dmaven.repo.local=$M2DIR -Dcuda.version=$CUDA_CLASSIFIER -DforceStdout -Dexpression=$1
+    mvn help:evaluate -q -pl $DIST_PL $MVN_URM_MIRROR -Prelease301 -Dmaven.repo.local=$M2DIR -Dcuda.version=$CUDA_CLASSIFIER -DforceStdout -Dexpression=$1
 }
 
 ART_ID=$(mvnEval project.artifactId)

